### PR TITLE
Feat(eos_cli_config_gen): Add support for MPLS resolution RIBs

### LINKED ISSUE #3415: Feat(eos_cli_config_gen): EVPN Address family support for tunnel-rib
### Enhancement summary

For the EVPN address family, we do not currently have support in the date model for configuration related to `next-hop mpls resolution ribs tunnel-rib`

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

For example, the below configuration can only be done via eos_cli field at the moment: 

```
router bgp <as>
   address-family evpn
     next-hop mpls resolution ribs tunnel-rib colored system-colored-tunnel-rib
```

### Describe the solution you would like

Extend the `router_bgp` data model to support such configuration 

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by chetryan**
## Change Summary

Add support for the `next-hop mpls resolution ribs` command.

Since the command supports both a Primary-RIB and a Fallback-RIB and both can be quite complex, I propose that we support string input for now. 

## Related Issue(s)

Fixes #3415 

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

- Updates the data model for eos_cli_config_gen
- Update the jinja templates for the documentation and eos config generation

## How to test

- `molecule test -s eos_cli_config_gen`

### User Checklist

- Review the change to the date modle
- Review the change to generated config and documentation

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/queue_monitor_length.md
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/queue_monitor_length.md:1]: You are deleting three unrelated files I think?

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/router-bgp-evpn-mpls.yml [hunk 1]
```diff
@@ -37,6 +37,9 @@ router_bgp:
     - ip_address: 192.168.255.2
       peer_group: EVPN-OVERLAY-PEERS
   address_family_evpn:
+    next_hop_mpls_resolution_ribs:
+      primary_rib: "tunnel-rib colored system-colored-tunnel-rib"
+      fallback_rib: "system-connected"
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/router-bgp-evpn-mpls.yml]: We try to avoid doing more of these now. Are the models very very complex? Do you have some example?

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/router-bgp-evpn-mpls.yml (hunk 1 above)
Comment (chetryan) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/router-bgp-evpn-mpls.yml]: I've attempted a different schema. It's the closest I can think of given current schema limitations.

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/queue_monitor_length.md
Comment (chetryan) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/queue_monitor_length.md:1]: I'm not sure how I messed this up - the files still show in my local git workspace.

Comment (chetryan) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/queue_monitor_length.md:1]: After re-writing history a few times it seems fixed now.

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/router-bgp-evpn-mpls.yml [hunk 2]
```diff
@@ -37,6 +37,10 @@ router_bgp:
     - ip_address: 192.168.255.2
       peer_group: EVPN-OVERLAY-PEERS
   address_family_evpn:
+    next_hop_mpls_resolution_ribs:
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/router-bgp-evpn-mpls.yml:40]: We support three levels here
```eos
router bgp 1
   address-family evpn
      next-hop mpls resolution ribs tunnel-rib test tunnel-rib colored system-colored-tunnel-rib system-connected
```

So I suggest primary_rib, secondary_rib, tertiary_rib.

Maybe make each of them a dict with `type: <str; system-connected | tunnel-rib | tunnel-rib-colored` and `rib_name`.
For now the name when type is tunnel-rib-colored can be hardcoded to the sytem one since that is the only available option.

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/router-bgp-evpn-mpls.yml:40]: Alternative have a key per option with `priority` 1-3 (or unset)

Comment (chetryan) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/router-bgp-evpn-mpls.yml:40]: Updated to use a list.. but still need to figure out the priority.

**Comment by ClausHolbechArista** [comment]
Please rebase and delete the .pickle files when it asks. We have removed those from the repo.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_bgp.schema.yml [hunk 3]
```diff
@@ -888,6 +888,25 @@ keys:
...
+              type: dict
+              keys:
+                rib_type:
+                  type: str
+                  required: true
+                  valid_values:
+                    - "system-connected"
+                    - "tunnel-rib-colored"
+                    - "tunnel-rib"
+                  description: "Type of RIB, one of system-connect|tunnel-rib-colored|tunnel-rib. For 'tunnel-rib', use 'rib_name' to specify the name of the Tunnel-RIB to use. Use 'tunnel-rib-colored' for 'tunnel-rib colored system-colored-tunnel-rib'"
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_bgp.schema.yml]: We document the choices under the key, so we don't have to repeat in the description.
Also I believe the colored is clear enough from the choice.

```suggestion
                  description: "Type of RIB. For 'tunnel-rib', use 'rib_name' to specify the name of the Tunnel-RIB to use."
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-bgp.j2 [hunk 4]
```diff
@@ -526,6 +526,17 @@
...
 {%         endif %}
+{%         if router_bgp.address_family_evpn.next_hop_mpls_resolution_ribs is arista.avd.defined() %}
+{%             set rib_position = ["Primary", "Secondary", "Tertiary" ] %}
+{%             for rib in router_bgp.address_family_evpn.next_hop_mpls_resolution_ribs %}
+{%                 if rib.rib_type == "tunnel-rib-colored" %}
+{%                     set rib_type = "tunnel-rib colored system-colored-tunnel-rib" %}
+{%                 else %}
+{%                     set rib_type = rib.rib_type %}
+{%                 endif %}
+- Next-hop MPLS resolution {{ rib_position[loop.index - 1] }}-RIB : {{ rib_type }} {{ rib.rib_name | arista.avd.default('') }}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-bgp.j2]: I think you have to append the rib_name in a separate `elif` above. Otherwise you will add an extra space at the end of the generated CLI/doc.

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-bgp.j2]:
```suggestion
- Next-hop MPLS resolution {{ rib_position[loop.index0] }}-RIB : {{ rib_type }} {{ rib.rib_name | arista.avd.default('') }}
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2 [hunk 5]
```diff
@@ -572,6 +572,18 @@ router bgp {{ router_bgp.as }}
...
+{%             for rib in router_bgp.address_family_evpn.next_hop_mpls_resolution_ribs %}
+{%                 if rib.rib_type == "tunnel-rib-colored" %}
+{%                     set rib_type = "tunnel-rib colored system-colored-tunnel-rib" %}
+{%                 else %}
+{%                     set rib_type = rib.rib_type %}
+{%                 endif %}
+{%                 set ns.evpn_mpls_resolution_ribs = ns.evpn_mpls_resolution_ribs ~ ' ' ~ rib_type ~ ' ' ~ rib.rib_name | arista.avd.default('') %}
+{%             endfor %}
+      {{ ns.evpn_mpls_resolution_ribs }}
+{%         endif %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2]: PyAVD does not work well with namespaces like this, so I suggest refactoring to use the `do` function line below.
```suggestion
{%         if router_bgp.address_family_evpn.next_hop_mpls_resolution_ribs is arista.avd.defined %}
{%             set evpn_mpls_resolution_ribs = [] %}
{%             for rib in router_bgp.address_family_evpn.next_hop_mpls_resolution_ribs %}
{%                 if rib.rib_type is arista.avd.defined("tunnel-rib-colored") %}
{%                     do evpn_mpls_resolution_ribs.append("tunnel-rib-colored system-colored-tunnel-rib" %}
{%                 elif rib.rib_type is arista.avd.defined("tunnel-rib") and rib.rib_name is arista.avd.defined %}
{%                     do evpn_mpls_resolution_ribs.append("tunnel-rib " ~ rib.rib_name
{%                 elif rib.rib_type is arista.avd.defined %}
{%                     do evpn_mpls_resolution_ribs.append(rib.rib_type) %}
{%                 endif %}
{%             endfor %}
{%             if evpn_mpls_resolution_ribs %}
      next-hop mpls resolution ribs {{ evpn_mpls_resolution_ribs | join(" ") }}
{%             endif %}
{%         endif %}
```

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

**Review by tgodaA — APPROVED** [review]
LGTM
